crates/driver/src/logic/competition/auction/order.rs

```diff
@@ -0,0 +1,82 @@
+use {
+    crate::{
+        logic::{competition, eth},
+        util,
+    },
+    primitive_types::U256,
+};
+
+/// An order in the auction.
+#[derive(Debug)]
+pub struct Order {
+    pub uid: competition::OrderUid,
+    pub from: eth::Address,
+    pub receiver: Option<eth::Address>,
+    pub valid_to: util::Timestamp,
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub side: competition::Side,
+    pub fee: Fee,
+    pub kind: Kind,
+    pub app_data: competition::AppData,
+    /// A partial order doesn't require the full amount to be traded. E.g. only
+    /// 10% of the requested amount may be traded, if this leads to the most
+    /// optimal solution.
+    pub partial: bool,
+    /// The autopilot marks orders as mature after a certain time period. The
+    /// solvers can use heuristics on this field to optimize solution sizes.
+    pub mature: bool,
+    pub executed: U256,
+    /// The onchain calls necessary to fulfill this order. These are set by the
+    /// user and included in the settlement transaction.
+    pub interactions: Vec<eth::Interaction>,
+
+    pub sell_source: SellSource,
+    pub buy_destination: BuyDestination,
+
+    // TODO This is a placeholder, there should probably be a proper submodule for signing stuff
+    pub signature: i32,
```

Comment:
This is passed in. You shouldn't need to sign or verify signatures in the `driver` at al - you can just assume its some "bytes" that need to be encoded as-is.